test/context.js

```diff
@@ -0,0 +1,52 @@
+import render from '../src/jsx';
+import { h, createContext } from 'preact';
+import chai, { expect } from 'chai';
+import sinonChai from 'sinon-chai';
+chai.use(sinonChai);
+
+// tag to remove leading whitespace from tagged template literal
+function dedent([str]) {
+	return str.split( '\n'+str.match(/^\n*(\s+)/)[1] ).join('\n').replace(/(^\n+|\n+\s*$)/g, '');
+}
+
+describe('context', () => {
+	let renderJsx = (jsx, opts) => render(jsx, null, opts).replace(/ {2}/g, '\t');
+
+	it('should support createContext', () => {
+		const { Provider, Consumer } = createContext();
+		let rendered = renderJsx(
+			<Provider value="correct">
+				<Consumer>
+					{(value) => (
+						<section>
+							value is: {value}
+						</section>
+					)}
+				</Consumer>
+			</Provider>
+		);
+
+		expect(rendered).to.equal(dedent`
+			<section>value is: correct</section>
+		`);
+	});
+
+	it('should support falsy context value', () => {
```

Comment:
Would love to see another test for nested `Provider` components of the same `Context`. In that case always the nearest Provider value should be used. The implementation in this PR very likely already supports this, just a test is missing :slightly_smiling_face: 

```jsx
const { Provider, Consumer } = createContext();
<Provider value="foo">
  <Provider value="bar">
    <Consumer>
      {value => value === "bar"} // must be bar
    </Consumer>
  </Provider>
</Provider>
```